### PR TITLE
Fix deprecation and bump dependencies

### PR DESCRIPTION
**Pull request by alexgavrisco**

### CODE REVIEW

**Comment by alexgavrisco** [comment]
It seems that there's an issue with production build. May be same cause as described here https://dev.clojure.org/jira/browse/CLJS-2392

**Comment by Gonzih** [comment]
Sorry, cant merge this with a broken build.

**Comment by Gonzih** [comment]
Could you also please rebase this on top of current master? Commit `5471d25` should be after latest merge commit that is on master.

File: project.clj
Comment (Gonzih) [inline project.clj]: Try to downgrade cljs compiler back to `1.9.542` to see if this fixes the build.

**Review by Gonzih — CHANGES_REQUESTED** [review]

File: project.clj
Comment (alexgavrisco) [reply project.clj]: It works with `1.9.542`, I'm gonna try to find the root cause. It would be nice to get it working with `1.10.x` since it improved build time significantly.
If it's a regression of `1.10.x`, I'll downgrade cljs.

**Comment by Gonzih** [comment]
Sounds like a plan, good luck! :)

**Comment by tirkarthi** [comment]
As I could see from the build error could it be due to drop of support for JDK 7 in the latest binary release ? Issue : https://github.com/google/closure-compiler/issues/2672 . I am still confused over why OracleJDK 7 succeeds though. Feel free to ignore this if I am wrong.

**Comment by Gonzih** [comment]
Oh yeah, that might be related. Ugh in that case removing Java 7 from travis.yml might make sense.

**Comment by alexgavrisco** [comment]
I've left only oraclejdk8 in travis config.
Also, I've had to use additional externs files for node's process and some closure compiler warning (as somebody recommended in slack).
It works fine for me with both dev and release builds.

**Comment by alexgavrisco** [comment]
I've pushed another commit which disabled `process-shim` and allowed to drop node process extern file.
I didn't know about this option which caused the issue with `js/process`, https://clojurescript.org/reference/compiler-options#process-shim

**Review by Gonzih — APPROVED** [review]

**Comment by Gonzih** [comment]
Thanks

**Pull request merged**
